### ISSUE TITLE
'Hide' button should cause conversations to completely disappear visually, not just cover them with blank

### ISSUE THREAD

**Issue by ripper234** [zen-mode/Whatsapp-Zen-Mode#48]
**This issue has a $2000 bounty.**- **Bounty currently on hold****

Currently, the Hide feature covers up a conversation:

![image](https://user-images.githubusercontent.com/172282/116375243-c489a000-a817-11eb-8d12-2e0e70d15d3c.png)

This hides some info about the conversation, last message sent, message count etc... but does not hide the conversation. It still appears as one of the conversations in the list:

![image](https://user-images.githubusercontent.com/172282/116375377-e420c880-a817-11eb-83d6-492e8302942b.png)

![image](https://user-images.githubusercontent.com/172282/116376528-06671600-a819-11eb-9e9c-422979e4437b.png)

We would like the Hide to *completely* remove this conversation from the list, without leaving any blank spots. It should appear *as if* the user Archived this conversation (but without actually archiving it). The difference from archive is that incoming messages will not cause this conversation to pop back in, but rather it will stay hidden "forever" or until defined differently by Zen Mode.

We should add a new command titled "Hidden Conversations" on the Zen Mode menu (picture explaining what ZM Menu is). This will show a list of all the hidden conversations (exact spec tbd), and allow the user to permentantly Unhide a specific conversation.

Note - when the user currently hides a conversation, it might be hidden forever, or for a specific amount of time e.g. 8 hours (see #19 ). FYI.

**Comment by ripper234** [comment]
Known issue: notification sounds might still be heard for incoming messages to hidden contacts (should separate this to a separate issue).

**Comment by gitpaybot** [comment]
A bounty of *2000 usd* was added to this issue. See task on [GitPay](https://gitpay.me/#/task/459)

**Comment by itzharDev** [comment]
1. the first bug is that the click event not bound well to element
2. after fixing that - getHoverContact name or something like - not work - so I hardcoded specific contact name
3. hiding element function actually not doing the work
is that code is yours? if yes I can give u a shortcut to solve this issue. otherwise don't waste your time

**Comment by itzharDev** [comment]
https://user-images.githubusercontent.com/12482438/117211745-11660b80-ae02-11eb-9e66-8a780d01f577.mp4

**Comment by liatzionimorad** [comment]
@itzharDev  I am referring this to our developers to reply. The code is ours .

**Comment by ripper234** [comment]
@itzharDev if the code works generally and not just for the case you showed, please submit it as a pull request.
If it answers the requirements we will merge it and award you the bounty.

You can do it either directly if you want, or via the Gitpay platform that can hold the money and arbitrate as needed (with commission).

**Comment by nul0op** [comment]
@ripper234, i don't see the pr yet. is this bug still open ?

**Comment by liatzionimorad** [comment]
@nul0op  currently on hold

**Comment by liatzionimorad** [comment]
team estimation:
e think there should be two points in № 39, so we overestimated this task.
1. In fact, this is a redesigned Hide function.
2. Creation of UI-interface in WWA for displaying "hidden" contacts. And adding a new item to the Zen menu.

Task № 39 can fail. But the chances are good that the function will work. Because we have access to a high-level API.
It's hard to say now how much code will be cleaned up.

Added new points to the assessment of the task 39 = 48 questions from the github.
2 hours - set the fake archive flag to communicate using the high-level API.
11h - create an interface with hidden chats
5 Hours - How to Embed UI in WWA HTML
9h - Code for cleaning the old concealment system

1) The initial result of the team is 27 hours.
2) In our experience, the team estimation error is 30%.
27 x 1.3 = 35
3.) Our insurance rate is 30%.
35 x 1.3 = 46
Full optimistic score out of 27 pessimistic out of 46 hours.

The revaluation took into account the new position No. 2, the evaluation of the position No. 1 as a whole did not change.

 During the first 2-4 hours, we will understand if we can solve the task. If it doesn't work, we'll spend 2-4 hours and stop working.
